sdcm/cluster.py

```diff
@@ -1823,10 +1820,56 @@ def restart_node_with_resharding(self, murmur3_partitioner_ignore_msb_bits=12):
             logger.debug('Resharding has been finished successfully (murmur3_partitioner_ignore_msb_bits={})'.
                          format(murmur3_partitioner_ignore_msb_bits))
 
-    def get_cql_auth(self):
-        user = self.params.get('authenticator_user', default=None)
-        password = self.params.get('authenticator_password', default=None)
-        return (user, password) if user and password else None
+    def _gen_nodetool_cmd(self, sub_cmd, args, options):
+        credentials = self.parent_cluster.get_db_auth()
+        if credentials:
+            options += '-u {} -pw {} '.format(*credentials)
+        return "nodetool {options} {sub_cmd} {args}".format(**locals())
+
+    def run_nodetool(self, sub_cmd, args="", options="", ignore_status=False):
+        """
+            Wrapper for nodetool command.
+            Command format: nodetool [options] command [args]
+
+        :param sub_cmd: subcommand like status
+        :param args: arguments for the subcommand
+        :param options: nodetool options:
+            -h	--host	Hostname or IP address.
+            -p	--port	Port number.
+            -pwf	--password-file	Password file path.
+            -pw	--password	Password.
+            -u	--username	Username.
+        :param ignore_status: don't throw exception if the command fails
+        :return: Remoter result object
+        """
+        cmd = self._gen_nodetool_cmd(sub_cmd, args, options)
+        result = self.remoter.run(cmd, ignore_status=ignore_status)
```

Comment:
`verbose=True` is used in executed some nodetool command for outputing the result in log, but the new function doesn't support this.

Is it expected?

Reply:
verbose is True by default. So any command without verbose=False, will be logged.

Reply:
You are right.